main.go

```diff
@@ -245,6 +245,9 @@ This is equivalent to setting PULUMI_MISSING_DOCS_ERROR=${! VALUE}.`)
 	cmd.PersistentFlags().StringVar(&context.PRTitlePrefix, "pr-title-prefix", "",
 		`The prefix to insert in the generated pull request title.`)
 
+	cmd.PersistentFlags().BoolVar(&context.DryRun, "dry-run", false,
+		`If true, don't actually create any PRs.`)
```

Comment:
Interesting.  I love removing side-effects and maybe even would argue the tool should not mess with PRs at all but given the history this is a great second-best.

Does the tool have any other observable side-effects besides creating the PR? Or everything else is scoped to your Git staging area? E.g. should dry-run turn off something else too?